iscript/src/iscript/mac.py

```diff
@@ -1202,44 +1191,45 @@ async def notarize_behavior(config, task):
 
     path_attrs = ["app_path"]
 
-    all_paths = get_app_paths(config, task)
-    langpack_apps = filter_apps(all_paths, fmt="autograph_langpack")
-    if langpack_apps:
-        await sign_langpacks(config, sign_config, langpack_apps)
-        all_paths = filter_apps(all_paths, fmt="autograph_langpack", inverted=True)
+    non_langpack_apps = []
+    for app in get_app_paths(config, task):
+        if fmt := get_langpack_format(app):
+            await sign_langpacks(config, sign_config, [app], fmt)
+        else:
+            non_langpack_apps.append(app)
```

Comment:
This probably looks more scary than it is., but ultimately the same things happen: we end up with all the non-langpacks in `non_langpack_apps`, and we sign any langpacks by the end of this block.

All of the other behaviours in here and `hardened_sign.py` are applying the same changes as this.